File: backend/src/main/java/com/culinaryheaven/domain/comment/dto/response/CommentsResponse.java
package com.culinaryheaven.domain.comment.dto.response;

import com.culinaryheaven.domain.comment.domain.Comment;
import io.swagger.v3.oas.annotations.media.Schema;

import java.util.ArrayList;
import java.util.List;

public record CommentsResponse(

        @Schema(description = "댓글 정보")
        List<CommentResponse> comments

) {

    public static CommentsResponse of(List<Comment> comments) {
        List<CommentResponse> commentResponses = comments
                .stream()
                .map(CommentResponse::of)
                .toList();

        return new CommentsResponse(commentResponses);
    }

}

File: android/data/src/main/java/com/jae464/data/remote/model/response/CommentsResponse.kt
package com.jae464.data.remote.model.response

import com.jae464.domain.model.Comment
import kotlinx.serialization.Serializable

@Serializable
data class CommentsResponse(
    val comments: List<CommentResponse>
)

fun CommentsResponse.toDomain(): List<Comment> {
    return comments.map { it.toDomain() }
}
